spec/routes/parameters.spec.ts

```diff
@@ -0,0 +1,261 @@
+import { OperationObject, PathItemObject } from 'openapi3-ts';
+import { z, ZodSchema } from 'zod';
+import { OpenAPIGenerator, RouteConfig } from '../../src';
+import { createTestRoute, registerSchema, testDocConfig } from '../lib/helpers';
+
+describe('parameters', () => {
+  it('generates a query parameter for route', () => {
+    const routeParameters = generateParamsForRoute({
+      request: { query: z.object({ test: z.string() }) },
+    });
+
+    expect(routeParameters).toEqual([
+      {
+        in: 'query',
+        name: 'test',
+        required: true,
+        schema: {
+          type: 'string',
+        },
+      },
+    ]);
+  });
+
+  it('generates a path parameter for route', () => {
+    const routeParameters = generateParamsForRoute({
+      request: { params: z.object({ test: z.string() }) },
+    });
+
+    expect(routeParameters).toEqual([
+      {
+        in: 'path',
+        name: 'test',
+        required: true,
+        schema: {
+          type: 'string',
+        },
+      },
+    ]);
+  });
+
+  it('generates a header parameter with array for route', () => {
+    const routeParameters = generateParamsForRoute({
+      request: {
+        headers: z.object({ test: z.string() }),
+      },
+    });
+
+    expect(routeParameters).toEqual([
+      {
+        in: 'header',
+        name: 'test',
+        required: true,
+        schema: {
+          type: 'string',
+        },
+      },
+    ]);
+  });
+
+  it('generates a header parameter with object for route', () => {
```

Comment:
This is the new test. The one above has just been renamed by adding "with array"